File: automation-test-playwright/pages/core/CustomerShellPage.ts
import { BasePage } from "./BasePage";

export class CustomerShellPage extends BasePage {
  protected readonly path?: string;

  /**
   * Opens the page.
   */
  async open(): Promise<void> {
    if (!this.path) {
      throw new Error("This customer page does not define a path.");
    }

    await this.visit(this.path);
  }

  async openHome(): Promise<void> {
    await this.visit("/customer/home");
  }

  /**
   * Navigates to contracts.
   */
  async goToContracts(): Promise<void> {
    await this.visit("/customer/contract/list");
  }

  /**
   * Navigates to invoices.
   */
  async goToInvoices(): Promise<void> {
    await this.visit("/customer/invoice/list");
  }

  /**
   * Navigates to buildings.
   */
  async goToBuildings(): Promise<void> {
    await this.visit("/customer/building/list");
  }

  /**
   * Navigates to transactions.
   */
  async goToTransactions(): Promise<void> {
    await this.visit("/customer/transaction/history");
  }

  /**
   * Navigates to services.
   */
  async goToServices(): Promise<void> {
    await this.visit("/customer/service");
  }

  /**
   * Navigates to profile.
   */
  async goToProfile(): Promise<void> {
    await this.visit("/customer/profile");
  }
}

File: automation-test-selenium/src/main/java/org/suntower/pages/core/CustomerShellPage.java
package org.suntower.pages.core;

import org.openqa.selenium.WebDriver;
import org.suntower.core.BasePage;

public class CustomerShellPage extends BasePage {
  protected String path;

  public CustomerShellPage(WebDriver driver) {
    super(driver);
  }

  public void open() {
    if (path == null || path.isBlank()) {
      throw new IllegalStateException("This customer page does not define a path.");
    }
    visit(path);
  }

  public void openHome() {
    visit("/customer/home");
  }

  public void goToContracts() {
    visit("/customer/contract/list");
  }

  public void goToInvoices() {
    visit("/customer/invoice/list");
  }

  public void goToBuildings() {
    visit("/customer/building/list");
  }

  public void goToTransactions() {
    visit("/customer/transaction/history");
  }

  public void goToServices() {
    visit("/customer/service");
  }

  public void goToProfile() {
    visit("/customer/profile");
  }
}
